File: 3 course/6 semester/Software development for mobile platforms/films_viewer_kotlin/app/src/main/java/com/example/films_viewer_kotlin/models/UserProfile.kt
package com.example.films_viewer_kotlin.models

data class UserProfile(
    val id: String,
    val email: String,
    var firstName: String,
    var lastName: String,
    var birthDate: String,
    var phone: String,
    var address: String,
    var country: String,
    var city: String,
    var bio: String,
    var gender: String
) {

    fun toFirestore(): Map<String, Any> {
        return mapOf(
            "email" to email,
            "firstName" to firstName,
            "lastName" to lastName,
            "birthDate" to birthDate,
            "phone" to phone,
            "address" to address,
            "country" to country,
            "city" to city,
            "bio" to bio,
            "gender" to gender
        )
    }

    companion object {
        fun empty(userId: String, email: String = ""): UserProfile {
            return UserProfile(
                id = userId,
                email = email,
                firstName = "",
                lastName = "",
                birthDate = "",
                phone = "",
                address = "",
                country = "",
                city = "",
                bio = "",
                gender = "Не указан"
            )
        }

        fun fromFirestore(data: Map<String, Any?>, id: String): UserProfile {
            return UserProfile(
                id = id,
                email = data["email"] as? String ?: "",
                firstName = data["firstName"] as? String ?: "",
                lastName = data["lastName"] as? String ?: "",
                birthDate = data["birthDate"] as? String ?: "",
                phone = data["phone"] as? String ?: "",
                address = data["address"] as? String ?: "",
                country = data["country"] as? String ?: "",
                city = data["city"] as? String ?: "",
                bio = data["bio"] as? String ?: "",
                gender = data["gender"] as? String ?: "Не указан"
            )
        }
    }
}

File: 3 course/6 semester/Software development for mobile platforms/films_viewer_swift/films_viewer_swift/films_viewer_swift/Models/UserProfile.swift
import Foundation

struct UserProfile {
    let id: String
    let email: String
    var firstName: String
    var lastName: String
    var birthDate: String
    var phone: String
    var address: String
    var country: String
    var city: String
    var bio: String
    var gender: String
    
    init(id: String, data: [String: Any]) {
        self.id = id
        self.email = data["email"] as? String ?? ""
        self.firstName = data["firstName"] as? String ?? ""
        self.lastName = data["lastName"] as? String ?? ""
        self.birthDate = data["birthDate"] as? String ?? ""
        self.phone = data["phone"] as? String ?? ""
        self.address = data["address"] as? String ?? ""
        self.country = data["country"] as? String ?? ""
        self.city = data["city"] as? String ?? ""
        self.bio = data["bio"] as? String ?? ""
        self.gender = data["gender"] as? String ?? "Не указан"
    }
    
    func toFirestore() -> [String: Any] {
        return [
            "email": email,
            "firstName": firstName,
            "lastName": lastName,
            "birthDate": birthDate,
            "phone": phone,
            "address": address,
            "country": country,
            "city": city,
            "bio": bio,
            "gender": gender
        ]
    }
    
    static func empty(userId: String, email: String = "") -> UserProfile {
        return UserProfile(id: userId, data: [
            "email": email,
            "firstName": "",
            "lastName": "",
            "birthDate": "",
            "phone": "",
            "address": "",
            "country": "",
            "city": "",
            "bio": "",
            "gender": "Не указан"
        ])
    }
}
